lib/src/html_input_stream.dart

```diff
@@ -236,15 +236,15 @@ class HtmlInputStream {
   /// including any character in 'characters' or EOF.
   String charsUntil(String characters, [bool opposite = false]) {
     final start = _offset;
-    String c;
-    while ((c = peekChar()) != null && characters.contains(c) == opposite) {
+    String? c;
+    while ((c = peekChar()) != null && characters.contains(c!) == opposite) {
       _offset += c.codeUnits.length;
     }
 
     return String.fromCharCodes(_chars.sublist(start, _offset));
   }
 
-  void unget(String ch) {
+  void unget(String? ch) {
     // Only one character is allowed to be ungotten at once - it must
     // be consumed again before any further call to unget
     if (ch != null) {
```

Comment:
The `ch.codeUnits.length` below is just a very inefficient way to write `ch.length`.

Reply:
I'll handle this in #141 as well.